lib/authz/serialized_project_rules_update_workflow.go

```diff
@@ -0,0 +1,638 @@
+package authz
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	iam_v2 "github.com/chef/automate/api/interservice/authz/v2"
+	"github.com/chef/automate/lib/cereal"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var errCanceled = errors.New("Canceled")
+
+// SerializedProjectUpdateWorkflowExecutor is a workflow that allows running 1
```

Comment:
we already had a type of workflow for serialized tasks, `SerializedWorkflowExecutor`, right? How is this one different?

Reply:
You might be thinking of the chain workflow executor. The problem with that is you need to know exactly how many things are in a chain. Here, we want to gather a list of all the things to be done before doing them so we can report progress in a more granular fashion. Also, i never anticipated that we would do chain -> parallel -> chain in terms of gluing together workflows, so thats not a thing that would just work